homeassistant/components/elmax/binary_sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax sensor platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
+
+        # Otherwise, add all the entities we found
+        entities = []
+        for zone in panel_status.zones:
+            entity = ElmaxSensor(
+                panel=coordinator.panel_entry,
+                elmax_device=zone,
+                panel_version=panel_status.release,
+                coordinator=coordinator,
+            )
+            if entity.unique_id not in known_devices:
+                entities.append(entity)
+        async_add_entities(entities, True)
+        known_devices.update([e.unique_id for e in entities])
+
+    # Register a listener for the discovery of new devices
+    coordinator.async_add_listener(_discover_new_devices)
+
+    # Immediately run a discovery, so we don't need to wait for the next update
+    _discover_new_devices()
+
+
+class ElmaxSensor(ElmaxEntity, BinarySensorEntity):
+    """Elmax Sensor entity implementation."""
+
```

Comment:
```suggestion
    
    _attr_device_class = BinarySensorDeviceClass.DOOR
    
```

Reply:
I appreciate your suggestion, but as stated in the [documentation](https://developers.home-assistant.io/docs/core/entity/#entity-class-or-instance-attributes), both forms are allowed. In my personal opinion, the proxy getter is a bit more explicit and readable.

Reply:
It seems the preference from the core developers are to use the _attr, especially for static things like in this case directly on the class.

Reply:
That's true, I've noticed that. But unless they change the statement in the docs (or at least clearly state that in the docs/best practices) I'd rather prefer proxy methods. But thank you guys for the help in reviewing this code, I appreciate that.

Reply:
we don't prefer proxy methods and in general spent time on cleaning those up. Would be nice if you saved us the cleanup effort.

I would have made the same request as done by @gjohansson-ST.